src/lib/support/ThreadOperationalDataset.cpp

```diff
@@ -142,58 +139,56 @@ class ThreadTLV final
 
     ThreadTLV * GetNext() { return reinterpret_cast<ThreadTLV *>(static_cast<uint8_t *>(GetValue()) + GetLength()); }
 
-    static bool IsValid(ByteSpan aData)
-    {
-        const uint8_t * const end = aData.data() + aData.size();
-        const uint8_t * curr      = aData.data();
-
-        while (curr + sizeof(ThreadTLV) < end)
-        {
-            const ThreadTLV * tlv = reinterpret_cast<const ThreadTLV *>(curr);
-
-            if (tlv->GetLength() == kLengthEscape)
-            {
-                break;
-            }
-
-            curr = reinterpret_cast<const uint8_t *>(tlv->GetNext());
-        }
-
-        return curr == end;
-    }
-
 private:
     uint8_t mType;
     uint8_t mLength;
 };
 
-bool OperationalDataset::IsValid(ByteSpan aData)
-{
-    return ThreadTLV::IsValid(aData);
-}
+/// OperationalDatasetView
 
-CHIP_ERROR OperationalDataset::Init(ByteSpan aData)
+bool OperationalDatasetView::IsValid(ByteSpan aData)
 {
-    if (aData.size() > sizeof(mData))
+    VerifyOrReturnValue(aData.size() <= kSizeOperationalDataset, false);
+
+    const ThreadTLV * tlv = reinterpret_cast<const ThreadTLV *>(aData.begin());
+    const ThreadTLV * end = reinterpret_cast<const ThreadTLV *>(aData.end());
+    while (tlv != end)
     {
-        return CHIP_ERROR_INVALID_ARGUMENT;
+        VerifyOrReturnValue(tlv + 1 <= end, false);                               // out of bounds
```

Comment:
The old IsValid looks like it did a `<` test here (i.e. disallowed things with `GetLength() == 0`).  Is the behavior change here intentional?

Reply:
The old logic was
```
while (curr + sizeof(ThreadTLV) < end) { ... }
return curr == end;
```
So an empty dataset is valid. This also make sense in terms of the invariant that `IsValid(mData)` remains true, and we start with an empty dataset.